{PROJECT_NAME}/Sources/Data/NetworkAPI/Core/RequestConfiguration.swift

```diff
@@ -2,38 +2,34 @@
 //  RequestConfiguration.swift
 //
 
-import Alamofire
 import Foundation
+import Moya
 
-protocol RequestConfiguration {
+enum RequestConfiguration {
```

Comment:
`RequestConfiguration` was a protocol to define a blueprint for every request. It couldn't be an enum or struct like this since we will have a specific config for each group of APIs based on the endpoint, e.g `user` -> `UserRequestConfiguration`.

It's also great if we keep `RequestConfiguration` as its responsibility with `Alamofire`, which means it's like a wrapper of Moya. Let's say we will not use Moya and change to another one in the future, so we don't need to update every "RequestConfiguration" to conform to the new lib. Conversely, we're going to import Moya to every request config to be able conform `TargetType` (and yeah, now it should be named `ABCTarget`).

Reply:
@vnntsu I am a bit confused about this.. Do you suggest to keep the `RequestConfiguration` protocol unchanged?

Reply:
@vnntsu We're using Moya, so I think using Enum will be fine. In case we have different groups of APIs, we can create another RequestConfiguration like UserRequestConfiguration, as long as we implement the protocol TargetType

Reply:
@Shayokh144 I suggest keeping `RequestConfiguration` protocol as a wrapper of TargetType; otherwise, we remove it. 
The enum `RequestConfiguration` is redundancy. We will have specific request configs such as `UserRequestConfiguration` or `OtherRequestConfiguration`, etc.

Reply:
Before Moya, our `RequestConfiguration` is `TargetType`, and it's a wrapper for Alamofire as we are moving to `Moya`, so I prefer to remove `RequestConfiguration`.

@vnntsu did raise a good point about changing to another networking library in the feature. We might need to update every `RequestConfiguration` to conform to the new library. In this case, I prefer to have a typealias:

```swift
typealias TargetType = Moya.TargetType
```

Let's say if we decide to remove Moya and go back to Alamofire, then we just need to implement the protocol `TargetType` (with baseURL, method...) in the network layer.

P.s: The reason I suggest `typealias TargetType = Moya.TargetType` instead of `typealias RequestConfiguration = Moya.TargetType` because it's more clear, using `RequestConfiguration` might confuse developers when we are using Moya.

Reply:
@suho typealias added and RequestConfiguration is deleted